src/protocol/friendTech/friendTech.ts

```diff
@@ -0,0 +1,188 @@
+import { Hex } from 'viem';
+import { EventLogTopics, Transaction } from '../../types';
+import { ABIs, FRIEND_TECH_ADDRESSES } from './constants';
+import { decodeTransactionInput, decodeLog } from '../../helpers/utils';
+
+export const contextualize = (transaction: Transaction): Transaction => {
+  const isFriendTech = detect(transaction);
+  if (!isFriendTech) return transaction;
+
+  return generate(transaction);
+};
+
+export const detect = (transaction: Transaction): boolean => {
+  /** implement your detection logic */
+  if (transaction.to !== FRIEND_TECH_ADDRESSES || !transaction.logs) {
+    return null;
+  }
+  // buyShares(address sharesSubject, uint256 amount)
+  if (transaction.sigHash === '0x6945b123') {
+    return true;
+  }
+
+  // sellShares(address sharesSubject, uint256 amount)
+  if (transaction.sigHash === '0xb51d0534') {
+    return true;
+  }
+
+  return false;
+};
+
+// Contextualize for mined txs
```

Comment:
```suggestion
// Contextualize for txs
```
We shouldn't keep using "mined" in these comments. It makes less sense with proof of stake